File: front/src/app/interfaces/user-profile.dto.ts
import { TopicDto } from './topic.dto';

export interface UserProfileDTO {
  username: string;
  email: string;
  abonnements: TopicDto[];
}

export interface UserProfileUpdateDTO {
  username?: string;
  email?: string;
  password?: string;
  desabonnements?: number[];
}

File: back/src/main/java/com/openclassrooms/mddapi/dto/UserProfileDto.java
package com.openclassrooms.mddapi.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileDto {
    private String username;
    private String email;
    private Set<TopicDto> abonnements;
}